Python/pylifecycle.c

```diff
@@ -563,6 +563,13 @@ pycore_init_runtime(_PyRuntimeState *runtime,
     if (_PyStatus_EXCEPTION(status)) {
         return status;
     }
+    
+    struct _gilstate_runtime_state *gilstate = &runtime->gilstate;
+
+    if (PyThread_tss_create(&gilstate->autoTSSkey) != 0) {
```

Comment:
This wrong, since _PyGILState_Init() will create a new key.

I prefer to split _PyGILState_Init() in two parts: see my PR #24819 fix.

Reply:
> This wrong, since _PyGILState_Init() will create a new key.
I think there is no problem.
after create PyThread_tss_create(&gilstate->autoTSSkey),
 autoTSSkey will set flag  key->_is_initialized = 1. 

when PyThread_tss_create(&gilstate->autoTSSkey) again,  just return 0. not create a new key.

```
    /* If the key has been created, function is silently skipped. */
    if (key->_is_initialized) {
        return 0;
    }
```
```
int
PyThread_tss_create(Py_tss_t *key)
{
    assert(key != NULL);
    /* If the key has been created, function is silently skipped. */
    if (key->_is_initialized) {
        return 0;
    }

    int fail = pthread_key_create(&(key->_key), NULL);
    if (fail) {
        return -1;
    }
    key->_is_initialized = 1;
    return 0;
}
```

Reply:
Oh. I didn't know that the Python _Py_tss_t structure has an _is_initialized member.

Well, I have another concern with your change. You are treating the gilstate as part of runtime. Well, technically, it's correct. But I'm trying to the GIL into the interpreter somehow, so I prefer to only create the GIL in pycore_create_interpreter().

The gilstate API is still a weird thing which is currently incompatible with subinterpreters: https://bugs.python.org/issue15751

Maybe fixing https://bugs.python.org/issue40522 will allow remove gilstate and rewrite the GILState APIs using the regular functions to get the current interpreter and Python thread state. For now, I prefer to ignore the gilstate API, it's rarely used in Python ;-)